content/articles/javafx-editable-table-view/index.md

```diff
@@ -0,0 +1,442 @@
+- Create another function named `loadData()` that creates dummy data for the application through iteration. Copy-paste
+  the following code into the file.
+
+```java
+    private void loadData(){
+        data_table=FXCollections.observableArrayList();
+
+        for(int x=1;x< 12;x++){
+
+        /* Generates the data items in the table */
+        data_table.add(new User(String.valueOf(x),"name "+x,"email "+x,"notes "+x,new Button("update")));
+        }
+
+        table_info.setItems(data_table);
+        }
+```
+
+- Run the application through the 'Main' class. The output looks like this:
+
+![Uneditable TableView](uneditable-tableview.png "Uneditable TableView")
+
+- Try to double-click on a cell and see if the contents can be modified.
+
+As noted, it can only select it but can't allow editing of the contents.
+
+### Make the Table cells editable
+
+Now in the Controller, do the following:
+
+- Add a function, that is the `editableCols()` method to make the cell have a 'Text Field's' property that not only allows displaying the message but also editing properties. It also changes the value of the cell once a commit is done. A commit is done when the value is changed and the Enter button is pressed.
+
+```java
+    private void editableCols(){
+        column_id.setCellFactory(TextFieldTableCell.forTableColumn());
+        column_id.setOnEditCommit(e->e.getTableView().getItems().get(e.getTablePosition().getRow()).setId(e.getNewValue()));
+
+        column_name.setCellFactory(TextFieldTableCell.forTableColumn());
+        column_name.setOnEditCommit(e->e.getTableView().getItems().get(e.getTablePosition().getRow()).setName(e.getNewValue()));
+
+        column_email.setCellFactory(TextFieldTableCell.forTableColumn());
+        column_email.setOnEditCommit(e->e.getTableView().getItems().get(e.getTablePosition().getRow()).setEmail(e.getNewValue()));
+
+        column_notes.setCellFactory(TextFieldTableCell.forTableColumn());
+        column_notes.setOnEditCommit(e->e.getTableView().getItems().get(e.getTablePosition().getRow()).setNotes(e.getNewValue()));
+
+        /* Allow for the values in each cell to be changable */
+        table_info.setEditable(true);
+```
+
+- To use the above function, call it inside the `initializeCols()` function, under the `col_update` line using the line below:
+
+```java
+editableCols();
+```
+
+- Re-run the application and try again. Notice that now the updates happen when the changes are done and the Enter key is pressed to commit the changes.
+On pressing the Update button, the outputs are sent to the terminal as configured before.
+
+![Final Result](final-output.png "Final Result")
+
+The content updates can be fetched for updating the content in the database.
+
+****
+
+### Conclusion
+
+At this point, the reader has learned and known how to:
+
+- Start a JavaFX project
+- Table creation and design
+- Adding dummy data to the table
+- Make the Table data editable
+- Fetch the contents edited in the console
+
```

Comment:
Add a link to the GitHub repo containing full code

Reply:
Check it out if it's all working well!!!

Reply:
@srishilesh

Reply:
Sure. Will do!